pkg/apiserver/api/v1/account.proto

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package kubecarrier.api.v1;
+option go_package = "v1";
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+import "meta.proto";
+import "types.proto";
+
+message Account {
+  ObjectMeta metadata = 1;
+  AccountSpec spec = 2;
+  AccountStatus status = 3;
+}
+
+message AccountSpec {
+  AccountMetadata metadata = 1;
+  repeated string roles = 2;
```

Comment:
Can this be an enum instead of string?

Reply:
There is some discussion around enum/string in this PR and also #421 . I am not a big fan of enum in protobuf, since it is really hard to extend in the future, it may cause issues if you add more values to it in the future, the different versions between client and server can lead to issues which are hard to debug.
Also, if you check the kube-apiserver, they also don't use enum and use strings.

Now I am thinking about if I should put the string to a dedicated message, for example:
```
message Role {
string type =1; 
}
```
In this way, at least it is typed I think? WDYT?

Reply:
> since it is really hard to extend in the future

what makes you think it's hard to extend in the future?

https://medium.com/@akhaku/protobuf-definition-best-practices-87f281576f31 (under 6) is a good suggestion for how to handle the unknown enum fields. 

And yeah, my biggest issue with naked meaningful strings is that they are both untyped and their allowed values aren't documented within the .proto file. What I like about enums is they're both typed and enumerate all possible valid states.

In this case, either enum or leave it as a string, creating an additional type looks like overkill for now. 

The biggest enum drawback is if the client encounters `UNKNOWN_ROLE` it cannot display it to the user in a meaningful way, aside from that I don't see issues with it.

Reply:
Kubernetes decided against Enums in their APIs - and I would stay with that for our API:
- https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md
- https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md
(Search for `enum`)

Reply:
ok, this doesn't change the fact we have enum like strings in our API. Could we at least document in the `.proto` files their expected values?